### PR TITLE
Some small fixes

### PR DESCRIPTION
**Pull request by meowthink**
See commit messages.

### CODE REVIEW

File: wechat/avatar.py [hunk 1]
```diff
@@ -55,8 +55,13 @@ def get_avatar(self, username):
                     img_file = os.path.join(self.avt_dir, filename)
                     if os.path.exists(img_file):
                         return Image.open(img_file)
+                    elif os.path.exists(img_file + ".bm"):
+                        img = self.read_bm(img_file + ".bm")
+                        return Image.fromarray(img)
                     else:
-                        return None
+                        pos, size = self.query_index(filename)
+                        return self.read_img(pos, size)
```
Comment (ppwwyyxx) [inline wechat/avatar.py]: Why do you add this? query_index does not work when there is no avatar database.

File: wechat/res.py [hunk 2]
```diff
@@ -186,10 +188,13 @@ def get_jpg_b64(img_file):
                 return None
             if not img_file.endswith('jpg') and \
                imghdr.what(img_file) != 'jpeg':
-                im = Image.open(open(img_file, 'rb'))
-                buf = cStringIO.StringIO()
-                im.convert('RGB').save(buf, 'JPEG', quality=JPEG_QUALITY)
-                return base64.b64encode(buf.getvalue())
+                 buf = cStringIO.StringIO()
```
Comment (ppwwyyxx) [inline wechat/res.py]: These lines are not correctly indented.

File: wechat/render.py [hunk 3]
```diff
@@ -153,6 +154,12 @@ def fallback():
         elif msg.type == TYPE_WX_VIDEO:
             # TODO: fetch video from resource
             return fallback()
+        elif msg.type == TYPE_VIDEO_FILE:
+            # TODO: fetch video from resource
```
Comment (ppwwyyxx) [inline wechat/render.py]: remove this comment

File: wechat/avatar.py (hunk 1 above)
Comment (meowthink) [reply wechat/avatar.py]: There're definitely situations need this, e.g. wechat updated from old version to new version. In such situation, both  avatar.index and png(.bm) are present, and unfortunately each one just cover a part.
There's a thing to concern here: If an avatar exists in both, we should prefer avatar.index first. But this little hack will in opposite.

File: wechat/render.py (hunk 3 above)
Comment (meowthink) [reply wechat/render.py]: Ok. Shall we put a TODO somewhere? As you may try out, video rendering is not so well, as I just did a trick letting fancybox dealing them as img.

**Comment by ppwwyyxx** [comment]
Video file and avatars ending with ".bm" extensions are supported already. Thanks for helping.

**Pull request closed without merging**